### DISCUSSION TITLE
Problem with Point_stat execution via Singularity Build

### DISCUSSION THREAD

**Discussion by justintsu** [dtcenter/METplus#2319] [Configuration] [answered]
I am attempting to run point_stat from a singularity container (singularity version 3.7.1-1).  The version of MET I am using is 11.0.1 and I run point_stat with the following command:

singularity exec ${PATH_TO_SING}/met_v11.0.1.simg  point_stat PYTHON_NUMPY ${PATH_TO_OBS}/sfc_2023070112.nc PointStat2023070112NEST1a0121x0121  -v 5  -outdir ./out/point_stat/SYNOP  -log ./DEBUGGER.log  -obs_valid_beg 20010101  -obs_valid_end 20230809

My goal is just to get point_stat running and operational in its simplest form before moving onto more complicated configuration file options. 

Running the command above produces the following error:

**DEBUG 5: Parsed observation field:
VarInfoGrib::dump():
  PTV       = -9999
  Code      = -9999
  LvlType   = -9999
  PCode     = -9999
  Center    = -9999
  Subcenter = -9999
ERROR  :
ERROR  : Dictionary::lookup_int() -> lookup failed for name "output_flag.seeps_mpr"
ERROR  :**

My Config file contains python embedding to read in my forecast fields, which are in IEEE BE format.  I've attached my configuration file as well as all of the data needed to replicate this problem.  If I missed anything please let me know and I'd be happy to post the missing data

[POINTSTATPROB_2.zip](https://github.com/dtcenter/METplus/files/12413616/POINTSTATPROB_2.zip)

**Comment by j-opatz** [comment, marked as answer]
Hello,

And thank you for your question. I was able to download and unpack your zip file, but I am unable to complete any kind of run with the data you sent due to a missing file. It seems that your Python script is trying to find "datahd_sfc_000000_000000_1a2000x0001_2023070112_00000000_infofld" which was not included in the zip file.

However, I think the error you originally experienced may be a quick fix on your end. The error 

> ERROR :
ERROR : Dictionary::lookup_int() -> lookup failed for name "output_flag.seeps_mpr"
ERROR :

is a common error type to see when the MET tool requires a setting but can't find it in the configuration file that passed at runtime. In this case, Point-Stat is looking for two additional settings in the ouput_flag dictionary that were added in MET 11.0.0, those for `seeps` and `seeps_mpr`. It's odd that your copy of the configuration file didn't have those settings given you're using MET 11.0.1, but you may have removed/altered the lines yourself.

Try setting your output_flag dictionary to the following and see if it clears up the error:

```
output_flag = {
   fho    = NONE;
   ctc    = NONE;
   cts    = NONE;
   mctc   = NONE;
   mcts   = NONE;
   cnt    = BOTH;
   sl1l2  = NONE;
   sal1l2 = NONE;
   vl1l2  = NONE;
   val1l2 = NONE;
   vcnt   = BOTH;
   pct    = NONE;
   pstd   = NONE;
   pjc    = NONE;
   prc    = NONE;
   ecnt   = NONE; // Only for HiRA.
   rps    = NONE; // Only for HiRA.
   eclv   = NONE;
   mpr    = NONE;
   seeps = NONE;
   seeps_mpr = NONE;
}

**Reply by justintsu** [reply]
Thanks, I've applied the changes and now get this error.  I've also attached that datahd file my python reader was complaining about.

**DEBUG 5: Parsed thresholds:
DEBUG 5: Matched pair filter columns:     (nul)
DEBUG 5: Matched pair filter thresholds:  NA
DEBUG 5: Forecast categorical thresholds: NA
DEBUG 5: Observed categorical thresholds: NA
DEBUG 5: Forecast continuous thresholds:  NA
DEBUG 5: Observed continuous thresholds:  NA
DEBUG 5: Continuous threshold logic:      UNION
DEBUG 5: Forecast wind speed thresholds:  NA
DEBUG 5: Observed wind speed thresholds:  NA
DEBUG 5: Wind speed threshold logic:      UNION
DEBUG 4: ClimoCDFInfo::set_cdf_ta() -> Since "cdf_bins" = 1, no climatology CDF bins will be applied.
ERROR  :
ERROR  : Dictionary::lookup_thresh() -> threshold "seeps_p1_thresh" must start with a valid comparison operator (<, <=, ==, !=, >, >=, lt, le, eq, ne, gt, ge).
ERROR  :**

[datahd_sfc_000000_000000_1a2000x0001_2023070112_00000000_infofld.zip](https://github.com/dtcenter/METplus/files/12422654/datahd_sfc_000000_000000_1a2000x0001_2023070112_00000000_infofld.zip)

**Reply by j-opatz** [reply]
Another missed update, my apologies! A quick glance through the [latest version of the Point-Stat configuration file](https://github.com/dtcenter/MET/blob/main_v11.1/data/config/PointStatConfig_default) shows a threshold for a new setting, `seeps_p1_thresh`, is necessary. Here's the default you can copy into your config file:

```
seeps_p1_thresh = >=0.1&&<=0.85;
```

Hopefully that brings your config file up-to-date!

**Reply by justintsu** [reply]
thank you! I realized I had two instances of topo_mask defined, so applying your last change and removing the second instance of topo_mask gave me a working config file.  Appreciate the help
